src/main/java/com/synopsys/integration/alert/common/descriptor/config/ui/CommonDistributionUIConfig.java

```diff
@@ -23,33 +23,77 @@
 public class CommonDistributionUIConfig {
+    private final Logger logger = LoggerFactory.getLogger(CommonDistributionUIConfig.class);
     public static final String KEY_NAME = "channel.common.name";
     public static final String KEY_CHANNEL_NAME = "channel.common.channel.name";
     public static final String KEY_PROVIDER_NAME = "channel.common.provider.name";
     public static final String KEY_FREQUENCY = "channel.common.frequency";
 
+    private final ConfigurationAccessor configurationAccessor;
+
+    @Autowired
+    public CommonDistributionUIConfig(final ConfigurationAccessor configurationAccessor) {
+        this.configurationAccessor = configurationAccessor;
+    }
+
     // TODO call this method in controller for distributions
     public List<ConfigField> createCommonConfigFields(final Set<String> channelDescriptors, final Set<String> providerDescriptors) {
         final ConfigField name = TextInputConfigField.createRequired(KEY_NAME, "Name");
-        final ConfigField frequency = SelectConfigField.createRequired(KEY_FREQUENCY, "Frequency", Arrays.stream(FrequencyType.values()).map(type -> type.getDisplayName()).collect(Collectors.toList()));
+        final ConfigField frequency = SelectConfigField.createRequired(KEY_FREQUENCY, "Frequency", Arrays.stream(FrequencyType.values()).map(type -> type.getDisplayName()).collect(Collectors.toList()), this::validateJobName);
         final ConfigField channelName = SelectConfigField.createRequired(KEY_CHANNEL_NAME, "Channel Type", channelDescriptors);
         final ConfigField providerName = SelectConfigField.createRequired(KEY_PROVIDER_NAME, "Provider Type", providerDescriptors);
 
         return List.of(name, channelName, frequency, providerName);
     }
 
+    private Collection<String> validateJobName(final FieldValueModel fieldToValidate, final FieldModel fieldModel) {
+        final Collection<String> errors = new ArrayList<>(1);
+        final String descriptorName = fieldModel.getDescriptorName();
+        final String jobName = fieldToValidate.getValue().orElse(null);
+        if (StringUtils.isNotBlank(jobName)) {
+            try {
+                final List<ConfigurationModel> configurations = configurationAccessor.getConfigurationsByDescriptorName(descriptorName);
+                final Boolean foundDuplicateName = configurations.stream()
+                                                       .map(configurationModel -> configurationModel.getField(CommonDistributionUIConfig.KEY_NAME).orElse(null))
+                                                       .filter(configurationFieldModel -> (null != configurationFieldModel) && configurationFieldModel.getFieldValue().isPresent())
+                                                       .anyMatch(configurationFieldModel -> jobName.equals(configurationFieldModel.getFieldValue().get()));
+                if (foundDuplicateName) {
+                    errors.add("A distribution configuration with this name already exists.");
```

Comment:
Does this only check duplicates between descriptors? Is this the check we want to do? Or do we want to check all configurations of a specific context type? I don't think we want to check a single descriptor.